### PR TITLE
ECS for Windows

### PR DESCRIPTION
**Pull request by dannyleesmith**
##### Corresponding Issue(s) or trello card(s):

TBC

##### Summary of change(s):

- Clean up ECS module
- Add output for use with Windows userdata
- Add option to provide tags

##### Will the change trigger resource destruction or replacement? If yes, please provide justification:

No

##### Does this update/change involve issues with other external modules? If so, please describe the scenario.

No

##### If input variables or output variables have changed or has been added, have you updated the README?

Yes

##### Do examples need to be updated based on changes?

Yes and have been

##### Note to the PR requester about Closing PR's
Please message the person that opened the issue when auto closing it on slack, as well as any other stake holders of deep interest. Only close the issue if you believe that the issue is fully resolved with this PR.

#### This PR may auto close the issue associated with it. If you feel the issue is not resolved please reopen the issue.

### CODE REVIEW

**Comment by dannyleesmith** [comment]
4df20db added because tests were failing on what is presumed to be because it was trying to -/+ the cluster

File: modules/cluster/main.tf [hunk 1]
```diff
@@ -1,3 +1,5 @@
-resource "aws_ecs_cluster" "ecs-cluster" {
-  name = "${var.ecs_cluster_name}"
+resource "aws_ecs_cluster" "this" {
```
Comment (johnctitus) [inline modules/cluster/main.tf]: If this resource title needs to be changed, we should use a more descriptive title than `this`

File: modules/cluster/main.tf [hunk 2]
```diff
@@ -1,3 +1,5 @@
...
+  name = "${var.cluster_name}"
+
+  tags = "${var.tags}"
```
Comment (johnctitus) [inline modules/cluster/main.tf]: Can we update the tag assignment to always set the standard Rackspace tags, and merge in the custom tags provided via this variable.  Would also require adding an `environment` variable to the module.  Example: https://github.com/rackspace-infrastructure-automation/aws-terraform-clb/blob/master/main.tf#L49-L54

File: modules/cluster/examples/ecs.tf [hunk 3]
```diff
@@ -3,14 +3,14 @@ provider "aws" {
...
 module "ecs" {
-  source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-ecs//?ref=v0.0.2"
+  source = "git@github.com:rackspace-infrastructure-automation/aws-terraform-ecs//modules/cluster/?ref=v0.0.3"
 
   cluster_name = "MyCluster"
+
+  tags = {
+    Environment     = "Development"
+    ServiceProvider = "Rackspace"
+    Terraform       = "true"
```
Comment (johnctitus) [inline modules/cluster/examples/ecs.tf]: Based on proposed changes to tags, we could remove the `Environment` and `ServiceProvider` tags from the example.

File: modules/cluster/main.tf (hunk 1 above)
Comment (johnctitus) [reply modules/cluster/main.tf]: Also this change is the reason the check_destruction test is failing, as terraform sees this a a new resource and removes the old.

File: modules/cluster/main.tf (hunk 2 above)
Comment (dannyleesmith) [reply modules/cluster/main.tf]: Sure, I'll do this tomorrow

File: modules/cluster/main.tf (hunk 1 above)
Comment (dannyleesmith) [reply modules/cluster/main.tf]: Yeah I figured it was this that was breaking the check_destruction. As for `this` I disagree. It's an ecs module, it only creates one cluster, the resource is already described as an `aws_ecs_cluster`. I'll shoot you a link as to why I've written this up in this way.

**Pull request merged**